app/helpers/user_helper.rb

```diff
@@ -0,0 +1,45 @@
+module UserHelper
+  # rubocop:disable Style/GuardClause
+  def friend_button(current_user, user)
+    'You are friends' if friend?(current_user, user)
+  end
+
+  def friend_post(current_user, user)
+    if friend?(current_user, user)
+      render @posts
+    else
+      render @other_posts
+    end
+  end
+
+  def friendship_button(current_user, user)
+    friendship = Friendship.find_by(requestor_id: current_user.id,
+                                    requestee_id: user.id) || Friendship.find_by(requestor_id: user.id,
+                                                                                 requestee_id: current_user.id)
+    if friendship.nil? and current_user != user
+      button_to 'Invite to friendship', user_friendships_path(user[:id]),
+                params: { friendship: { requestee_id: user.id } }, class: 'button_to'
+    end
+  end
+
+  def friendship_update_button(current_user, user)
+    pending_request = Friendship.find_by(requestor_id: user.id, requestee_id: current_user.id, status: 'pending')
+    if current_user != user and pending_request
```

Comment:
There should be NO application logic in the views nor helpers (no Post.new, Comment.create, current_user.friends.find_by(...)...). Everything connected with data processing should be encapsulated in models and used in controllers.